capi-tilt/README.md

```diff
@@ -0,0 +1,141 @@
+```
+You should verify if the installation is correct:
+```
+tilt version
+```
+**Note:** 
+1. Tilt requires **Docker** to be installed as a non-root user and **kubectl** binary. It is assumed that this is already taken care of in your local environment. 
+2. Tilt also requires a working kubernetes cluster to work on. For this purpose we are using **Kind**.
+
+## Tiltfile
+The cluster api repository has a ```Tiltfile``` in the root directory. This ```Tiltfile``` contains all the instructions for running the dev environment. 
+
+1. To start with, do the following changes so that we use CAPD as the infrastructure provider: 
+
+```
+infrastructure_provider = DOCKER_PROVIDER
+#infrastructure_provider = AWS_PROVIDER
+```
+## Create tilt-settings.json
+
+Create a tilt-settings.json file and place it in your local copy of cluster-api. When using Kind, you’ll need a way to push your images to a registry to they can be pulled. Most users test with GCR, but you could also use something like Docker Hub. Here is an example:
+
+```
+{
+    "default_registry": "hub.docker.com/u/<user>",
+    "enable_providers": ["docker", "kubeadm-bootstrap", "kubeadm-control-plane"]
+}
+```
+
+## Kind cluster
+
+Because the Docker provider needs to access Docker on the host, a custom kind cluster configuration is required:
+
+``` 
+cat > kind-cluster-with-extramounts.yaml <<EOF
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+nodes:
+- role: control-plane
+  extraMounts:
+    - hostPath: /var/run/docker.sock
+      containerPath: /var/run/docker.sock
+EOF
+kind create cluster --config ./kind-cluster-with-extramounts.yaml
+``` 
+
+## Running the tilt environment
+
+Run the tilt environment from your local copy of cluster api repo using the following command:
+
+``` 
+tilt up 
+``` 
+
+This will open the command-line HUD as well as a web browser interface. You can monitor Tilt’s status in either location. After a brief amount of time, you should have a running development environment, and you should now be able to create a cluster.
+
+## Create cluster
+
+In this folder you can find example .yaml files to run simple cluster for testing cluster api. First you need to create a cluster itself:
+
+```
+kubectl apply -f cluster.yaml
+```
+
+## Create control plane
+
+Next create a control plane node:
+
+```
+kubectl apply -f controlplane.yaml
+```
+
+## Copy working cluster Kubeconfig
+
+After the control plane node is up, you have to retrieve and save the working cluster Kubeconfig:
+
+```
+kubectl --namespace=default get secret/my-cluster-kubeconfig -o json | jq -r .data.value | base64 --decode   > ./capi-quickstart.kubeconfig
+```
+
+## Install and patch Calico 
+
+In this example we deploy Calico CNI solution. Deploy Calico using **capi-quickstart.kubeconfig** copied in last step with:
+
+```
+kubectl --kubeconfig=./capi-quickstart.kubeconfig \
+  apply -f https://docs.projectcalico.org/v3.8/manifests/calico.yaml
```

Comment:
v3.8 is rather old.
have you tried v3.11 or v3.12?

Reply:
No I have not tried it. These lines is taken from: 

https://master.cluster-api.sigs.k8s.io/user/quick-start.html

and these steps follows cluster api books own workflow.